flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/translators/SinkTransformationTranslator.java

```diff
@@ -315,6 +315,13 @@ private <I, R> R adjustTransformations(
                         Transformation::getDescription,
                         Transformation::setDescription);
 
+                // handle coLocationGroupKey.
```

Comment:
Why change it in this PR?

Reply:
When I migrated the use of legacy discard sink in [StreamGraphCoLocationConstraintTest.java](https://github.com/apache/flink/pull/23013/commits/f39d85dc04bb54aae265fe769eac501243ce7c19#diff-e476bce2e644b45118686e9cbaf71fd25e0ed5949a2e82dc7c62172cb9a60d57) to the new implementation, I found that `SinkV2` did not properly handle coLocationGroup, so I also fixed this issue for migration. If you think this is worth a separate PR, I agree to split it out.